src/vs/workbench/contrib/editSessions/browser/editSessionsStorageService.ts

```diff
@@ -196,6 +206,30 @@ export class EditSessionsWorkbenchService extends Disposable implements IEditSes
 		return authenticationSession !== undefined;
 	}
 
+	private cachedMachines: Map<string, string> | undefined;
+
+	async getMachineById(machineId: string) {
+		await this.initialize(false);
```

Comment:
It seems possible to have `doInitialize` invoked multiple times as it's not a single promise that every operation awaits for, not sure if there is a real harm though, e.g., multiple auth requests.